src/main/java/org/testng/internal/BaseTestMethod.java

```diff
@@ -133,11 +115,21 @@ public ITestClass getTestClass() {
    */
   @Override
   public void setTestClass(ITestClass tc) {
-    assert null != tc;
-    if (! tc.getRealClass().equals(m_method.getDeclaringClass())) {
-      assert m_method.getDeclaringClass().isAssignableFrom(tc.getRealClass()) :
-        "\nMISMATCH : " + tc.getRealClass() + " " + m_method.getDeclaringClass();
+    if (tc == null) {
+      throw new IllegalArgumentException("test class cannot be null");
+    }
+    boolean sameClass = tc.getRealClass().equals(m_method.getDeclaringClass());
+    if (!sameClass) {
+      boolean assignable = m_method.getDeclaringClass().isAssignableFrom(tc.getRealClass());

```

Comment:
I think only the second check is useful and already covers the first check.

Reply:
Reviews are here to check mistakes. But sure, you made again another awesome job! :beer: